lib/type1.g

```diff
@@ -592,26 +585,40 @@ BIND_GLOBAL( "Objectify", function ( type, obj )
     if not IsType( type )  then
         Error("<type> must be a type");
     fi;
+    flags := FlagsType(type);
     if IsHPCGAP then
-        flags := FlagsType(type);
         if IS_LIST( obj )  then
-            if IS_SUBSET_FLAGS(flags, IsAtomicPositionalObjectRepFlags) then
+            if IS_SUBSET_FLAGS(flags, FLAGS_FILTER(IsAtomicPositionalObjectRep)) then
                 FORCE_SWITCH_OBJ( obj, FixedAtomicList(obj) );
             fi;
         elif IS_REC( obj )  then
             if IS_ATOMIC_RECORD(obj) then
-                if IS_SUBSET_FLAGS(flags, IsNonAtomicComponentObjectRepFlags) then
+                if IS_SUBSET_FLAGS(flags, FLAGS_FILTER(IsNonAtomicComponentObjectRep)) then
                     FORCE_SWITCH_OBJ( obj, FromAtomicRecord(obj) );
                 fi;
-            elif not IS_SUBSET_FLAGS(flags, IsNonAtomicComponentObjectRepFlags) then
+            elif not IS_SUBSET_FLAGS(flags, FLAGS_FILTER(IsNonAtomicComponentObjectRep)) then
                 FORCE_SWITCH_OBJ( obj, AtomicRecord(obj) );
             fi;
         fi;
     fi;
     if IS_LIST( obj )  then
+        #Assert(0, IS_SUBSET_FLAGS(flags, FLAGS_FILTER(IsPositionalObjectRep)));
+        Assert(0, not IS_SUBSET_FLAGS(flags, FLAGS_FILTER(IsComponentObjectRep)));
```

Comment:
Maybe disable this by default, but have it enabled e.g. in most test suites (as long as the use `START_TEST`)?
```suggestion
        #Assert(1, IS_SUBSET_FLAGS(flags, FLAGS_FILTER(IsPositionalObjectRep)));
        Assert(1, not IS_SUBSET_FLAGS(flags, FLAGS_FILTER(IsComponentObjectRep)));
```